packages/playwright-test/src/runner.ts

```diff
@@ -179,7 +179,10 @@ export class Runner {
 
   private async _run(list: boolean, testFileReFilters: FilePatternFilter[], projectNames?: string[]): Promise<FullResult> {
     const filesByProject = await this._collectFiles(testFileReFilters, projectNames);
-    return await this._runFiles(list, filesByProject, testFileReFilters);
+    const result = await this._runFiles(list, filesByProject, testFileReFilters);
+    if (this._didBegin)
+      await this._reporter.onEnd?.(result);
```

Comment:
- Let's move this to `_runAllTests` instead, because it has one more onEnd so it would be better to handle all onEnd calls in one place.
- A test would also be nice (just console.log from globalteardown and onEnd).

Reply:
done